### DISCUSSION TITLE
Exclude header and footer from first and last page

### DISCUSSION THREAD

**Discussion by danielveselinov** [spatie/laravel-pdf#31] [Ideas]
I think the possibility of excluding the header/footer from the first/last page is a great feature that can be implemented in this package since it could be used not only for generating invoices but also for much more complex PDFs.

**Comment by ArielMejiaDev** [comment]
You are able to generate two PDF files the first one with the portrait and another skipping the first page with:

```php
Pdf::view('test')
    ->withBrowsershot(function (Browsershot $browsershot) {
        $browsershot>initialPageNumber(8);
    });
```

and then use a library to merge files like: https://github.com/hanneskod/libmergepdf

```php
se iio\libmergepdf\Merger;

$portrait = Pdf::view('portrait')->save('portrait.pdf');
$content = Pdf::view('your-content')->save('content.pdf');

$merger = new Merger;
$merger->addIterator([$portrait, $content]);
$createdPdf = $merger->merge();
```

it should works

**Comment by greeto** [comment]
@danielveselinov did you manage to solve it somehow?

**Comment by Aamish26622** [comment]
@freekmurze is there a way in package for this issue? we are going live very soon and want this to be resolved before that

**Comment by michaelzon** [comment]
@greeto @Aamish26622 I managed to remove the footer from the first page like this: 
```
        use function Spatie\LaravelPdf\Support\pdf;

        $fooFilename = "foo__" . Str::uuid() . ".pdf";
        $booFilename = "boo__" . Str::uuid() . ".pdf";

        pdf()->view('foo')->disk('public')->save($fooFilename);
        pdf()->view('boo')->footerView('footer')->disk('public')->save($booFilename);

        $pdfs = [
            'foo' => Storage::disk('public')->path($fooFilename),
            'boo' => Storage::disk('public')->path($booFilename),
        ];

        $merger = new Merger;
        $merger->addFile($pdfs['foo']);
        $merger->addFile($pdfs['boo']);

        $createdPdf = $merger->merge();
        $fooBooFilename = "foo_boo__" . Str::uuid() . ".pdf";
        Storage::disk('public')->put($fooBooFilename, $createdPdf);
```
        
Only downside is that you save three files to storage when you actually only want to write one. But there is probably also a solution for that. If I found something I will post it here.
